gps/deduce.go

```diff
@@ -167,18 +168,49 @@ func (m bitbucketDeducer) deduceRoot(path string) (string, error) {
 	return "bitbucket.org" + v[2], nil
 }
 
+func getBitbucketRepo(s string) (string, error) {
+	ss := strings.Split(s, "/")
+	if len(ss) < 3 {
+		return "", errors.Errorf("bitbucket repo path %q appears malformed", s)
+	}
+
+	repoStr := ss[1] + "/" + ss[2]
+
+	if repoStr[len(repoStr)-4:] == ".git" {
+		repoStr = repoStr[:len(repoStr)-4]
+	}
+
+	if repoStr[len(repoStr)-3:] == ".hg" {
+		repoStr = repoStr[:len(repoStr)-3]
+	}
+
+	return repoStr, nil
+}
+
 func (m bitbucketDeducer) deduceSource(path string, u *url.URL) (maybeSources, error) {
 	v := m.regexp.FindStringSubmatch(path)
 	if v == nil {
 		return nil, fmt.Errorf("%s is not a valid path for a source on bitbucket.org", path)
```

Comment:
should we do `%q` here? in case the empty string gets passed in, it can help clarify what is actually happening.

Reply:
Let me take a look at the callers of this to see if that's a potential scenario. If so it wouldn't hurt, although I'd likely raise it as its own PR just to avoid too much change in this one.